docs/source/format/Versioning.rst

```diff
@@ -0,0 +1,71 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+Format Versioning and Stability
+===============================
+
+Starting with version 1.0.0 (not yet released), Apache Arrow utilizes
+**two versions** to describe each release of the project. These are
+the **Format Version** and the **Library Version**. Each Library
+Version has a corresponding Format Version, and multiple versions of
+the library may have the same format version. For example, library
+versions 2.0.0 and 3.0.0 may both track format version 1.0.0.
+
+For library versions prior to 1.0.0, major releases (e.g. 0.14.0 to
+0.15.0) may contain API changes. From 1.0.0 onward, we follow
```

Comment:
The example given here is actually a "minor" release and not a "major" as written. This terminology is always a bit confusing (as such releases are communicated as major), but given that below semver is mentioned (which defines those terms) and also "minor" and "major" or used to describe the format versions, using a different term here for library versions is extra confusing.